test/uk/gov/hmrc/hecstubs/controllers/AccountingPeriodControllerSpec.scala

```diff
@@ -0,0 +1,391 @@
+          }
+
+          "start date and CorrelationId  are invalid" in {
+
+            val expectedJson: JsValue =
+              badJsonResponseTwo(
+                "INVALID_START_DATE",
+                "INVALID_CORRELATIONID",
+                "Invalid query parameter startDate.",
+                "Invalid header CorrelationId."
+              )
+
+            val result: Future[Result] =
+              controller.accountingPeriod(validCtutr, inValidStartDate(0), validEndDate)(
+                fakeRequest("live", "correlationid")
+              )
+            status(result) shouldBe Status.BAD_REQUEST
+            contentAsJson(result) mustBe expectedJson
+          }
+
+          "end date and CorrelationId  are invalid" in {
+
+            val expectedJson: JsValue =
+              badJsonResponseTwo(
+                "INVALID_END_DATE",
+                "INVALID_CORRELATIONID",
+                "Invalid query parameter endDate.",
+                "Invalid header CorrelationId."
+              )
+
+            val result: Future[Result] =
+              controller.accountingPeriod(validCtutr, validStartDate, inValidEndDate(0))(
+                fakeRequest("live", "correlationid")
+              )
+            status(result) shouldBe Status.BAD_REQUEST
+            contentAsJson(result) mustBe expectedJson
+          }
+
+        }
+
+        "three invalid paramaters" when {
+
+          "ctutr, startDate and end date are invalid" in {
+
+            val expectedJson: JsValue =
+              badJsonResponseThree(
+                "INVALID_CTUTR",
+                "INVALID_START_DATE",
+                "INVALID_END_DATE",
+                "Invalid parameter ctutr.",
+                "Invalid query parameter startDate.",
+                "Invalid query parameter endDate."
+              )
+
+            val result: Future[Result] =
+              controller.accountingPeriod(inValidCtUtr(0), inValidStartDate(0), inValidEndDate(0))(
+                fakeRequest("live", UUID.randomUUID().toString)
+              )
+            status(result) shouldBe Status.BAD_REQUEST
+            contentAsJson(result) mustBe expectedJson
+
+          }
+
+          "ctutr, startDate and correlationId are invalid" in {
+
+            val expectedJson: JsValue =
+              badJsonResponseThree(
+                "INVALID_CTUTR",
+                "INVALID_START_DATE",
+                "INVALID_CORRELATIONID",
+                "Invalid parameter ctutr.",
+                "Invalid query parameter startDate.",
+                "Invalid header CorrelationId."
+              )
+
+            val result: Future[Result] =
+              controller.accountingPeriod(inValidCtUtr(0), inValidStartDate(0), validEndDate)(
+                fakeRequest("live", "correlationId")
+              )
+            status(result) shouldBe Status.BAD_REQUEST
+            contentAsJson(result) mustBe expectedJson
+          }
+        }
+
+        "four invlid paramters" when {
```

Comment:
```suggestion
        "four invalid paramters" when {
```

Reply:
I think i have fixed this in latest commit